gpage.h

```diff
@@ -222,16 +228,17 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		auto cmp = std::less<>{};
+		return !cmp(p, &storage[0]) && cmp(p, &storage[total_size - 1]);
```

Comment:
I assumed that was intentional, to avoid checking the "extra" location added to the end of allocations, though I didn't investigate whether there's any reason to do so. I tried to use `storage.get()` for "I want the address of the page" and `&storage[foo]` for "I want the address of this byte *within* the page" because I think the array notation is slightly easier to read. That obviously no longer applies when the upper bound is not within the page; I agree with your suggestion.